src/ert/parsing/lark_parser.py

```diff
@@ -231,17 +240,52 @@ def _tree_to_dict(
     return config_dict
 
 
+ArgPairList = List[Tuple[FileContextToken]]
+ParsedArgList = List[Union[FileContextToken, ArgPairList]]
+
+
 def _substitute_args(
-    args: List[Any], constraints: SchemaItem, defines: Defines
-) -> List[Any]:
+    args: ParsedArgList,
+    constraints: SchemaItem,
+    defines: Defines,
+) -> ParsedArgList:
     if constraints.substitute_from < 1:
         return args
-    return [
-        _substitute(defines, x, constraints.expand_envvar)
-        if i + 1 >= constraints.substitute_from and isinstance(x, FileContextToken)
-        else x
-        for i, x in enumerate(args)
-    ]
+
+    def substitute_arglist_tuple(
+        tup: Tuple[FileContextToken],
+    ) -> Tuple[FileContextToken]:
+        tuple_kw = tup[0]
+        tuple_args = tup[1:]
+        substituted_tuple_args = [
+            _substitute_token(defines, token, constraints.expand_envvar)
+            for token in tuple_args
+        ]
+
+        return tuple([tuple_kw] + substituted_tuple_args)
+
+    def substitute_arg(
+        i, arg: Union[FileContextToken, List[Tuple[FileContextToken]]]
+    ) -> Union[FileContextToken, List[Tuple[FileContextToken]]]:
+        can_be_substituted = (i + 1) >= constraints.substitute_from
+
+        if not can_be_substituted:
+            return arg
+
+        if isinstance(arg, FileContextToken):
+            return _substitute_token(defines, arg, constraints.expand_envvar)
+
+        if isinstance(arg, list):
+            # It is a list of keyword tuples
+            return [substitute_arglist_tuple(x) for x in arg]
+
+        raise ValueError(
+            f"Expected "
+            f"Union[FileContextToken, List[Tuple[FileContextToken]]], "
+            f"got {arg}"
+        )
+
+    return [substitute_arg(i, arg) for i, arg in enumerate(args)]
```

Comment:
I was thinking that maybe `return [substitute_arg(arg) if (i + 1) >= constraints.substitute_from else arg for i, arg in enumerate(args)]` might be a little easier to read.